gym/wrappers/rescale_observation.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+
+import gym
+from gym import spaces
+
+
+def rescale_values(values, old_low, old_high, new_low, new_high):
+    rescaled_values = new_low + (new_high - new_low) * (
+        (values - old_low) / (old_high - old_low))
+    rescaled_values = np.clip(rescaled_values, new_low, new_high)
+    return rescaled_values
+
+
+def verify_observation_space_type(observation_space):
+    if not isinstance(observation_space, spaces.Box):
+        raise TypeError("Expected Box observation space. Got: {}"
+                        "".format(type(observation_space)))
+
+
+def verify_observation_space_bounds(observation_space):
+    if np.any(~np.isfinite((
+            observation_space.low, observation_space.high))):
+        raise ValueError(
+            "Observation space 'low' and 'high' need to be finite."
+            " Got: observation_space.low={}, observation_space.high={}"
+            "".format(observation_space.low, observation_space.high))
+
+
+def rescale_box_space(observation_space, low, high):
+    shape = observation_space.shape
+    dtype = observation_space.dtype
+
+    new_low = low + np.zeros(shape, dtype=dtype)
+    new_high = high + np.zeros(shape, dtype=dtype)
+
+    observation_space = spaces.Box(
+        low=new_low, high=new_high, shape=shape, dtype=dtype)
+
+    return observation_space
+
+
+class RescaleObservation(gym.ObservationWrapper):
+    def __init__(self, env, low, high):
+        r"""Rescale observation space to a range [`low`, `high`].
+
+        For `Box` spaces, `low` and `high` can be either scalar or vector, and
+        will be broadcasted according to numpy broadcasting rules. For `Tuple`
+        and `Dict` spaces, both `low` and `high` are expected to be scalar.
```

Comment:
could you specify in the docstring that elements of Tuple or Dict space are expected to be Box?